jda/src/main/java/co/aikar/commands/JDACommandManager.java

```diff
@@ -161,6 +161,17 @@ public void registerCommand(BaseCommand command) {
         }
     }
 
+    public void unregisterCommand(BaseCommand command) {
```

Comment:
See how BukkitCommandManager does unregistering.

A single RootCommand can be used by multiple BaseCommands.
You need to remove all subcommands provided by the base command from the root command, and then only remove the root command when it has no subcommands left.